lib/core/src/Cardano/Wallet/DB/Wallets/Store.hs

```diff
@@ -0,0 +1,128 @@
+    , PersistAddressBook (..)
+    , blockHeaderFromEntity
+    -- * Testing
+    , mkStoreWallet
+    )
+    where
+
+import Cardano.Wallet.DB
+    ( ErrBadFormat (ErrBadFormatAddressPrologue) )
+import Cardano.Wallet.DB.Checkpoints.Store
+    ( PersistAddressBook (..), blockHeaderFromEntity, mkStoreCheckpoints )
+import Cardano.Wallet.DB.Sqlite.Schema
+    ( EntityField (CheckpointWalletId), Wallet )
+import Cardano.Wallet.DB.Wallets.State
+    ( DeltaWalletState
+    , DeltaWalletState1 (ReplacePrologue, UpdateCheckpoints)
+    , WalletState (WalletState)
+    )
+import Control.Exception
+    ( toException )
+import Control.Monad
+    ( forM )
+import Control.Monad.Except
+    ( ExceptT (ExceptT), runExceptT )
+import Data.DBVar
+    ( Store (Store, loadS, updateS, writeS) )
+import Data.DeltaMap
+    ( DeltaMap (Adjust, Delete, Insert) )
+import Data.Generics.Internal.VL
+    ( view, (^.) )
+import Database.Persist.Sql
+    ( Entity, SqlPersistT, deleteWhere, entityVal, selectList, (==.) )
+import Prelude
+
+import qualified Cardano.Wallet.Primitive.Types as W
+import qualified Data.Map.Strict as Map
+
+ {-------------------------------------------------------------------------------
+    WalletState Store
+-------------------------------------------------------------------------------}
+-- | Store for 'WalletState' of multiple different wallets.
+mkStoreWallets
+    :: forall s key. (PersistAddressBook s, key ~ W.WalletId)
+    => Store (SqlPersistT IO)
+        (DeltaMap key (DeltaWalletState s))
+mkStoreWallets = Store{loadS=load,writeS=write,updateS=update}
+  where
+    write = error "mkStoreWalletsCheckpoints: not implemented"
+
+    update _ (Insert wid a) =
+        writeS (mkStoreWallet wid) a
+    update _ (Delete wid) = do
+        -- FIXME LATER during ADP-1043:
+        --  Deleting an entry in the Checkpoint table
+        --  will trigger a delete cascade. We want this cascade
+        --  to be explicit in our code.
+        deleteWhere [CheckpointWalletId ==. wid]
+    update _ (Adjust wid da) =
+        updateS (mkStoreWallet wid) undefined da
+        -- FIXME LATER during ADP-1043:
+        --   Remove 'undefined'.
+        --   Probably needs a change to 'Data.DBVar.updateS'
+        --   to take a 'Maybe a' as parameter instead of an 'a'.
+
+    load = do
+        wids <- fmap (view #walId . entityVal) <$> selectAll
+        runExceptT $ do
+            xs <- forM wids $ ExceptT . loadS . mkStoreWallet
+            pure $ Map.fromList (zip wids xs)
+      where
+        selectAll :: SqlPersistT IO [Entity Wallet]
+        selectAll = selectList [] []
+
+-- | Store for 'WalletState' of a single wallet.
+mkStoreWallet
+    :: forall s. PersistAddressBook s
+    => W.WalletId
+    -> Store (SqlPersistT IO) (DeltaWalletState s)
+mkStoreWallet wid =
+    Store{ loadS = load, writeS = write, updateS = \_ -> update }
+  where
+    storeCheckpoints = mkStoreCheckpoints wid
+    -- storeTransactions = mkStoreTransactions wid
```

Comment:
```suggestion
    -- TODO: storeTransactions = mkStoreTransactions wid
```

Makes it more clear that it's not a piece of code that was commented out, but a piece that is to be written in.